src/python/pants/backend/shell/dependency_inference.py

```diff
@@ -52,10 +52,10 @@ class ShellMapping:
 
 @rule(desc="Creating map of Shell file names to Shell targets", level=LogLevel.DEBUG)
 async def map_shell_files() -> ShellMapping:
-    all_expanded_targets = await Get(Targets, AddressSpecs([DescendantAddresses("")]))
-    shell_tgts = tuple(tgt for tgt in all_expanded_targets if tgt.has_field(ShellSources))
+    all_targets = await Get(Targets, AddressSpecs([DescendantAddresses("")]))
+    shell_tgts = tuple(tgt for tgt in all_targets if tgt.has_field(ShellSourceField))
```

Comment:
The changes in this file mean that dep inference only happens for the file-based atomic targets, i.e. `ShellSourceTarget` and `ShellTestTarget`.